File: src/optical_element/MATLAB/Furion_toroid_Mirror.m
classdef Furion_toroid_Mirror<w_oe
    %波动光学圆柱镜  
    properties
        R1; %圆柱镜面曲率半径  
        R2;
        rho1;
        rho2;
    end
    
    methods
        function obj=Furion_toroid_Mirror( beam_in,ds,di,chi,theta,surface,R1,R2,rho1,rho2,grating)  
            obj = obj@w_oe( beam_in,ds,di,chi,theta,surface,grating) ;  %初始化父类  optical_element的参数      
            obj.R1 =R1;
            obj.R2=R2;
            obj.rho1=rho1;
            obj.rho2 =rho2;
            obj.reflect;%反射光束          
        end
    end   
    
      methods %光学追迹
        function tracing(obj)
           obj.g_mirror =  g_Furion_toroid_Mirror( obj.gbeam_in,obj.ds,obj.di,obj.chi,obj.theta,no_surfe(),obj.R1,obj.R2,obj.rho1,obj.rho2,grating)  ;
          
        end
      end
    
end

File: src/optical_element/Furion_toroid_Mirror.cpp
#include "Furion_toroid_Mirror.h"

using namespace Furion_NS;

Furion_Toroid_Mirror::Furion_Toroid_Mirror(Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, double r, Grating* grating)
    : W_Oe(beam_in, ds, di, chi, theta, surface, grating)
{
    cout << "Furion_Toroid_Mirror ³õÊ¼»¯" << endl;
    this->R1 = R1;
    this->R2 = R2;
    this->rho1 = rho1;
    this->rho2 = rho2;
}

Furion_Toroid_Mirror::~Furion_Toroid_Mirror()
{

}

void Furion_Toroid_Mirror::run(Beam* beam_in, double ds, double di, double chi, double theta, No_Surfe* surface, Grating* grating)
{
    cout << "Furion_Toroid_MirrorµÄrun" << endl;
    reflect(beam_in, ds, di, chi, theta);
}

std::string Furion_Toroid_Mirror::tracing()
{
    cout << "Furion_Toroid_Mirror" << endl;
    this->g_FT_mirror = new G_Furion_Toroid_Mirror(this->gbeam_in, this->ds, this->di, this->chi, this->theta, this->surface, this->R1, this->R2, this->rho1, this->rho2, this->grating);
    g_FT_mirror->run(this->gbeam_in, this->ds, this->di, this->chi, this->theta);

    return ("Furion_Toroid_Mirror");
}

